### PR TITLE
Raise exception in case of wrongly normalized target triple

### LINKED ISSUE #8709: Index out of bounds with bad target on compile
I ran this command:
```
crystal build --release --no-debug --cross-compile --target "x86_64" src/start_server.cr -o app
```
and got this error: (using crystal 0.31.1)

```
Index out of bounds (IndexError)
  from Crystal::Codegen::Target::new<String>:Crystal::Codegen::Target
  from ~procProc(String, Nil)@src/compiler/crystal/command.cr:415
  from OptionParser::ParseTask#process_double_flag<String, Proc(String, Nil), Bool>:Nil
  from OptionParser#parse<Array(String)>:Nil
  from Crystal::Command#create_compiler<String, Bool, Bool, Bool, Bool, Bool>:Crystal::Command::CompilerConfig
  from Crystal::Command#run:(Bool | Crystal::Compiler::Result | Nil)
  from main
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

I tried building from source to see if the issue was already fixed. I get this with 0.33.0-dev

```
Index out of bounds (IndexError)
  from /Users/jeremywoertink/Development/crystal/lang/src/indexable.cr:614:8 in '[]'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/codegen/target.cr:17:44 in 'initialize'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/codegen/target.cr:12:3 in 'new'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:435:13 in '->'
  from /Users/jeremywoertink/Development/crystal/lang/src/option_parser.cr:255:3 in 'process_double_flag'
  from /Users/jeremywoertink/Development/crystal/lang/src/option_parser.cr:238:9 in 'process_handler'
  from /Users/jeremywoertink/Development/crystal/lang/src/option_parser.cr:213:9 in 'parse'
  from /Users/jeremywoertink/Development/crystal/lang/src/option_parser.cr:185:5 in 'parse'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:58:5 in 'create_compiler'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:283:11 in 'create_compiler'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:169:14 in 'build'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:71:16 in 'run'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:49:5 in 'run'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal/command.cr:48:3 in 'run'
  from /Users/jeremywoertink/Development/crystal/lang/src/compiler/crystal.cr:8:1 in '__crystal_main'
  from /Users/jeremywoertink/Development/crystal/lang/src/crystal/main.cr:97:5 in 'main_user_code'
  from /Users/jeremywoertink/Development/crystal/lang/src/crystal/main.cr:86:7 in 'main'
  from /Users/jeremywoertink/Development/crystal/lang/src/crystal/main.cr:106:3 in 'main'
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

My guess is it's just because I passed a bad target. When I changed it to `--target "x86_64-linux-gnu"` it compiled fine.

### PR DESCRIPTION
**Pull request by Sija**
Fixes #8709

### CODE REVIEW

**Comment by Sija** [comment]
@RX14 ping

File: src/compiler/crystal/codegen/target.cr [hunk 1]
```diff
@@ -64,8 +67,6 @@ class Crystal::Codegen::Target
   def freebsd_version
     if @environment =~ /freebsd(\d+)\.\d+/
       $1.to_i
-    else
-      nil
```
Comment (RX14) [inline src/compiler/crystal/codegen/target.cr]: Please remove this unrelated cosmetic change.

Comment (Sija) [reply src/compiler/crystal/codegen/target.cr]: Is it really necessary?

Comment (RX14) [reply src/compiler/crystal/codegen/target.cr]: Is it really neccesary to include unrelated changes in PRs?

I write code like this because I think it's clearer and more explicit, not because I missed a trick.

Comment (Sija) [reply src/compiler/crystal/codegen/target.cr]: Ok, I've reverted this change.

**Comment by Sija** [comment]
GTG?

**Comment by straight-shoota** [comment]
Thanks @Sija
